### ISSUE TITLE
Rmd: Markdown fenced code blocks with language identifier cause `Error: Malformed file!`

### ISSUE THREAD

**Issue by salim-b** [jimhester/lintr#505]
## Problem

Consider the following reprex:

````r
example_file <- tempfile()

readr::write_file(path = example_file,
                  x = '# Rmd example file

## R code chunk

The following is linted fine:

```{r}
print ("hi" )
```

## Plain code block

The following fenced code block _without_ a language identifier is not linted (also fine!):

```
print ("hi" )
```

## Code block with language identifier

The language identifier of the following code block triggers a \'malformed file\' error (bad!):

```r
print ("hi" )
```
')

lintr::lint(filename = example_file)
#> Error: Malformed file!
````
<sup>Created on 2020-07-03 by the [reprex](https://reprex.tidyverse.org/) package (v0.3.0)</sup>

Markdown code blocks with a language identifier are valid R Markdown syntax and should simply be excluded from linting.

## Possibly related

If I understand it correctly, R Markdown files should be supported by lintr as of https://github.com/jimhester/lintr/commit/50ab86db4ebc7272d95990f8d15b3fabe0278936 fixing https://github.com/jimhester/lintr/issues/32.

In https://github.com/jimhester/lintr/issues/252, a similar issue was raised which has been solved by https://github.com/jimhester/lintr/commit/fdd19507481fddcc8ec200463c5b8c73bc58e949, _assumedly_ without properly handling language identifiers. I'm not familiar with lintr's internals and didn't dive into the code, but I guess the R code extraction logic in https://github.com/jimhester/lintr/blob/master/R/extract.R is erroneous. Maybe @russHyde or @jimhester could have a look at this?

Maybe also related: https://github.com/jimhester/lintr/issues/322

**Comment by russHyde** [comment]
could you confirm this error using the current master branch of lintr, we have a test case that covers the case:
````
```r
abc = 123
```
````

**Comment by salim-b** [comment]
> could you confirm this error using the current master branch of lintr

Just tested the current master branch: The issue is gone, no more error. :+1:  I should have tested this before opening this issue 🙈 Thanks for the hint!
